File: app/src/main/java/app/deadmc/devnetworktool/helpers/CheckHelper.java
package app.deadmc.devnetworktool.helpers;

/**
 * Created by Feren on 15.06.2016.
 */
public class CheckHelper {
    public static boolean isValidIp (String ip) {
        try {
            if ( ip == null || ip.isEmpty() ) {
                return false;
            }

            String[] parts = ip.split( "\\." );
            if ( parts.length != 4 ) {
                return false;
            }

            for ( String s : parts ) {
                int i = Integer.parseInt( s );
                if ( (i < 0) || (i > 255) ) {
                    return false;
                }
            }
            if ( ip.endsWith(".") ) {
                return false;
            }

            return true;
        } catch (NumberFormatException nfe) {
            return false;
        }
    }

    public static boolean isValidPort(int port) {
        if ((port <65536) && (port > 0))
            return true;
        else
            return false;
    }

    public static int portFromString(String portString) {
        int port = 80;
        if (!portString.isEmpty()) {
            try {
                port = Integer.parseInt(portString.toString());
            } catch (Exception e) {}
        }
        return port;
    }

}

File: app/src/main/java/app/deadmc/devnetworktool/helpers/CheckHelper.kt
package app.deadmc.devnetworktool.helpers

fun isValidIp(ip: String?): Boolean {
    try {
        if (ip == null || ip.isEmpty()) {
            return false
        }

        val parts = ip.split("\\.".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        if (parts.size != 4) {
            return false
        }

        for (s in parts) {
            val i = Integer.parseInt(s)
            if (i < 0 || i > 255) {
                return false
            }
        }
        return !ip.endsWith(".")
    } catch (nfe: NumberFormatException) {
        return false
    }

}

fun isValidPort(port: Int): Boolean = port in 0..65536

fun portFromString(portString: String): Int {
    var port = 80
    if (!portString.isEmpty()) {
        safe {
            port = Integer.parseInt(portString)
        }
    }
    return port
}


